File: client-scalaz/src/main/scala/org/openapitools/client/api/AssetListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AssetListResponse._

case class AssetListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[AssetResponse]],
hasMoreResults: Option[Boolean],
count: Option[Integer],
returning: Option[String])

object AssetListResponse {
  import DateTimeCodecs._

  implicit val AssetListResponseCodecJson: CodecJson[AssetListResponse] = CodecJson.derive[AssetListResponse]
  implicit val AssetListResponseDecoder: EntityDecoder[AssetListResponse] = jsonOf[AssetListResponse]
  implicit val AssetListResponseEncoder: EntityEncoder[AssetListResponse] = jsonEncoderOf[AssetListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/asset_list_response.clj
(ns sirqul-io-t-platform.specs.asset-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.asset-response :refer :all]
            )
  (:import (java.io File)))


(def asset-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of asset-response-spec)
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :count) int?
   (ds/opt :returning) string?
   })

(def asset-list-response-spec
  (ds/spec
    {:name ::asset-list-response
     :spec asset-list-response-data}))
